service/src/request_keys.rs

```diff
@@ -0,0 +1,93 @@
+/*
+	Copyright 2021 Integritee AG and Supercomputing Systems AG
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+*/
+
+//! Request state keys from a fellow validateer.
+
+use crate::enclave::tls_ra::enclave_request_key_provisioning;
+use codec::Error as CodecError;
+use futures::executor;
+use itc_rpc_client::direct_client::{
+	DirectApi, DirectClient as DirectWorkerApi, Error as DirectRpcClientError,
+};
+use itp_api_client_extensions::PalletTeerexApi;
+use itp_enclave_api::remote_attestation::TlsRemoteAttestation;
+use itp_types::ShardIdentifier;
+use log::*;
+use sgx_types::sgx_quote_sign_type_t;
+use std::string::String;
+use substrate_api_client::ApiClientError;
+
+#[derive(Debug, thiserror::Error)]
+enum Error {
+	#[error("ApiClient Error: {0}")]
+	ApiClient(#[from] ApiClientError),
+	#[error("{0}")]
+	Codec(#[from] CodecError),
+	#[error("Could not fetch any data.")]
+	EmptyValue,
+	#[error("{0}")]
+	JsonRpSeeClient(#[from] jsonrpsee::types::Error),
+	#[error("{0}")]
+	Serialization(#[from] serde_json::Error),
+	#[error("{0}")]
+	DirectRpcClient(#[from] DirectRpcClientError),
+}
```

Comment:
Is this due to copy-pasting, or the error intentionally declared?

Reply:
intentionally, as preparatory step to move this to its own crate. But that's an overkill for this PR, hence removed it. Thanks for pointing it out!